proposals/awsvpc-task-metadata.md

```diff
@@ -0,0 +1,40 @@
+<!--
+Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You may
+not use this file except in compliance with the License. A copy of the
+License is located at
+
+     http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+-->
+
+
+### Introduction 
+All containers launched with `awsvpc` networking mode get a local IPv4 address from the `169.254.172.0/22`. This can be used to identify a container and the task that it belongs to based on its IPv4 address. A HTTP endpoint exposed by the ECS agent can in turn serve metadata to containers based on the request's IPv4 address. 
```

Comment:
It's worth noting somewhere that this relies on the `awsvpc` implementation details for Linux and may not be applicable for Windows (once we actually get `awsvpc` working there).

Reply:
Anirudh and I talked about this last week. We think awsvpc implementation for Windows will work exactly the same way. The mechanism to record the IP address->container assignment might slightly differ, but agent will still use the source IP address to identify the task.

Reply:
Awesome!